src/components/Charts/ClientChart.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import { Chart } from '@/components/Charts/Chart';
+import { useCountry } from '@/domain/contexts/CountryContext';
+import { CountryData } from '@/domain/entities/country/CountryData';
+
+export function ClientChart() {
+  const { getCountryData } = useCountry();
+  const [data, setData] = useState<CountryData | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const countryData = await getCountryData(50);
+        setData(countryData);
```

Comment:
maybe the context could store and distribute the returned data, so that we don't have to fetch it every time. especially if this data is not likely to change too often (and I assume it doesn't, since it's historical data. or will we have fcs data for the future, as well?